### PR TITLE
fix: objects with functions cant be equal

### PR DESCRIPTION
**Pull request by pspaczek**
## Description
equal() has a problem comparing object with functions. To solve this issue I use "JSON" trick to remove functions

## Related Issue

Closes #

## Checklist:

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

**Review by AlanLes — APPROVED** [review]
lgtm 👍

**Review by kubajmarek — COMMENTED** [review]
Could this be applied directly to `equal` function? There are no other uses of `equal()` except in those examples that were changed.

**Review by kubajmarek — COMMENTED** [review]
I know that performance probably isn't the biggest concern here, but it feels a bit wrong to me to use `fast-equals` but at the same parse & stringify each input. What's the exact cause of the issue?

File: src/components/atoms/UiCheckbox/UiCheckbox.vue [hunk 1]
```diff
@@ -172,7 +172,7 @@ const checkboxId = computed(() => (props.id || `checkbox-${uid()}`));
 const isChecked = computed(() => {
   if (Array.isArray(props.modelValue)) {
     return !!props.modelValue.find((option) => (
-      equal(props.value, option)));
+      equal(JSON.parse(JSON.stringify(props.value)), JSON.parse(JSON.stringify(option)))));
```
Comment (kubajmarek) [inline src/components/atoms/UiCheckbox/UiCheckbox.vue:175]: If the problem is on top layer, `Object.assign({}, target)` or `{ ...target }` are significantly more performant at "plain"ing the object.

Comment (pspaczek) [reply src/components/atoms/UiCheckbox/UiCheckbox.vue:175]: Nope, the problem is deeper. In for example 'suffixAttrsProps' and `onClick` function

**Comment by kubajmarek** [comment]
How about [this](https://github.com/lukeed/dequal) package? 🤔

**Comment by pspaczek** [comment]
> fast-equals

sometimes we need to add a function (for example `onClick`) to props for deeper components. For example, we want to add `onClick` to UiMultipleAnswer to the info button. Checkbox uses an object with 'onClick' as a value. Without these changes, we equal return false

**Review by sonya0504 — APPROVED** [review]

**Comment by pspaczek** [comment]
@kubajmarek fast-equals "can compare" `Functions` too. But they compare references :(, they do not convert it `.toString()`

**Pull request merged**